gtsam/hybrid/DCMarginals.h

```diff
@@ -0,0 +1,32 @@
+/* ----------------------------------------------------------------------------
+ * Copyright 2020 The Ambitious Folks of the MRG
+ * See LICENSE for the license information
+ * -------------------------------------------------------------------------- */
+
+/**
+ *
```

Comment:
I don’t remember we had added this, @varunagrawal ?

Reply:
Yup, I added this. This is point 4 in the PR comment: since DCValues and DCMarginals are structs in DC_types.h, I just moved them to their own file so as to be consistent with the rest of GTSAM.

I took a bit of design liberty here, so I hope that's okay?

Reply:
Same comment about authorship...

Reply:
This file has Kevin as the sole author as per your recommendation. I only added myself as author to `testDCMarginals.cpp` since I wrote that file from scatch.